### PR TITLE
tardigrade: Update sso-admin module to use `alternate_identifier` instead of deprecated `filter`

### PR DESCRIPTION
**Pull request by deanp3it**
updated versions.tf to use 4.4.0 for AWS, updated main.tf of account-assignment module to use alternate_identifier instead of filter

### CODE REVIEW

**Comment by deanp3it** [comment]
@lorengordon how do you want this incremented in bumpversion?

File: modules/account-assignment/main.tf [hunk 1]
```diff
@@ -2,7 +2,7 @@ resource "aws_ssoadmin_account_assignment" "this" {
   instance_arn       = local.sso_instance_arn
   permission_set_arn = var.account_assignment.permission_set_arn != null ? var.account_assignment.permission_set_arn : data.aws_ssoadmin_permission_set.this[0].arn
 
-  principal_id   = var.account_assignment.principal_type == "GROUP" ? data.aws_identitystore_group.this[0].id : data.aws_identitystore_user.this[0].id
+  principal_id   = var.account_assignment.principal_type == "GROUP" ? data.aws_identitystore_group.this.alternate_identifier : data.aws_identitystore_user.this.alternate_identifier
```
Comment (lorengordon) [inline modules/account-assignment/main.tf:5]: I feel like this shouldn't change? What was the rationale for this change?

```suggestion
  principal_id   = var.account_assignment.principal_type == "GROUP" ? data.aws_identitystore_group.this[0].id : data.aws_identitystore_user.this[0].id
```

File: modules/account-assignment/main.tf [hunk 2]
```diff
@@ -54,6 +54,6 @@ data "aws_ssoadmin_instances" "this" {
 
 locals {
   # Reduce api calls
-  identity_store_id = var.account_assignment.identity_store_id != null ? var.account_assignment.identity_store_id : data.aws_ssoadmin_instances.this[0].identity_store_ids[0]
-  sso_instance_arn  = var.account_assignment.instance_arn != null ? var.account_assignment.instance_arn : data.aws_ssoadmin_instances.this[0].arns[0]
+  identity_store_id = var.account_assignment.identity_store_id != null ? var.account_assignment.identity_store_id : data.aws_ssoadmin_instances.this.identity_store_ids[0]
+  sso_instance_arn  = var.account_assignment.instance_arn != null ? var.account_assignment.instance_arn : data.aws_ssoadmin_instances.this.arns[0]
```
Comment (lorengordon) [inline modules/account-assignment/main.tf:58]: Similarly, not sure why this is changing?

```suggestion
  identity_store_id = var.account_assignment.identity_store_id != null ? var.account_assignment.identity_store_id : data.aws_ssoadmin_instances.this[0].identity_store_ids[0]
  sso_instance_arn  = var.account_assignment.instance_arn != null ? var.account_assignment.instance_arn : data.aws_ssoadmin_instances.this[0].arns[0]
```

**Review by lorengordon — CHANGES_REQUESTED** [review]

File: versions.tf [hunk 3]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 4.30"
+      version = ">= 4.4.0"
```
[the remark this answers is not in the document]
Comment (deanp3it) [reply versions.tf]: Fixed

**Review by lorengordon — COMMENTED** [review]
Let's go ahead and make this a major version bump. Don't forget the changelog!

**Comment by deanp3it** [comment]
@lorengordon the error message I was fixing was fixed, still failing because of the API but I think it should be ready to merge.

**Review by lorengordon — APPROVED** [review]

**Pull request merged**